app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/EligibilityController.scala

```diff
@@ -135,23 +136,18 @@ class EligibilityController @Inject() (
     implicit val eori: EORI = request.eoriNumber
     emailService.retrieveEmailByEORI(eori) flatMap {
       _.emailType match {
-        case EmailType.VerifiedEmail =>
-          store.get[EligibilityJourney].map {
-            case Some(journey) =>
-              val form =
-                journey.eoriCheck.value.fold(eoriCheckForm)(eoriCheck =>
-                  eoriCheckForm.fill(FormValues(eoriCheck.toString))
-                )
-              Ok(checkEoriPage(form, eori, routes.EligibilityController.getCustomsWaivers().url))
-            case _ => handleMissingSessionData("Eligibility journey")
-          }
-        case EmailType.UnVerifiedEmail =>
-          Redirect(routes.UpdateEmailAddressController.updateUnverifiedEmailAddress()).toFuture
-        case EmailType.UnDeliverableEmail =>
-          Redirect(routes.UpdateEmailAddressController.updateUndeliveredEmailAddress()).toFuture
+        case VerifiedEmail =>
+          // User may have been redirected here from ECC so ensure the EligibilityJourney is present if not already.
+          store.getOrCreate[EligibilityJourney](EligibilityJourney()).map { journey =>
```

Comment:
Should we create all the journeys which we are doing at Account Controller, just to be safe?

Reply:
The UndertakingJourney is already created if not present by the UndertakingController. All of this would probably benefit from another pass so it's consistent.